src/sentry/static/sentry/app/components/events/groupingInfo/groupingVariant.tsx

```diff
@@ -0,0 +1,183 @@
+        if (variant.values) {
+          data.push([t('Fingerprint values'), variant.values]);
+        }
+        break;
+      case EventGroupVariantType.SALTED_COMPONENT:
+        component = variant.component;
+        if (variant.values) {
+          data.push([t('Fingerprint values'), variant.values]);
+        }
+        if (variant.config?.id) {
+          data.push([t('Grouping Config'), variant.config.id]);
+        }
+        break;
+      default:
+        break;
+    }
+
+    if (component) {
+      data.push([
+        t('Grouping'),
+        <GroupingTree key={component.id}>
+          <GroupingComponent
+            component={component}
+            showNonContributing={this.state.showNonContributing}
+          />
+        </GroupingTree>,
+      ]);
+    }
+
+    return [data, component];
+  }
+
+  renderTitle() {
+    const {variant} = this.props;
+    const isContributing = variant.hash !== null;
+
+    return (
+      <Tooltip
+        title={isContributing ? t('Contributing variant') : t('Non-contributing variant')}
+      >
+        <GroupVariantTitle>
+          <ContributionIcon isContributing={isContributing} />
+          {t('By')}{' '}
+          {variant.description
+            .split(' ')
+            .map(i => capitalize(i))
+            .join(' ')}
+        </GroupVariantTitle>
+      </Tooltip>
+    );
+  }
+
+  renderContributionToggle() {
+    const {showNonContributing} = this.state;
+
+    return (
+      <ButtonBar merged active={showNonContributing ? 'all' : 'relevant'}>
+        <Button barId="relevant" size="xsmall" onClick={this.handleHideNonContributing}>
+          {t('Contributing values')}
+        </Button>
+        <Button barId="all" size="xsmall" onClick={this.handleShowNonContributing}>
+          {t('All values')}
+        </Button>
+      </ButtonBar>
+    );
+  }
+
+  render() {
+    const [data, component] = this.getVariantData();
+
+    return (
+      <GroupVariantListItem>
+        <Header>
+          {this.renderTitle()}
+          {hasNonContributingComponent(component) && this.renderContributionToggle()}
+        </Header>
+
+        <KeyValueList data={data} isContextData isSorted={false} />
+      </GroupVariantListItem>
+    );
+  }
+}
+
+const GroupVariantListItem = styled('li')`
+  margin-bottom: ${space(4)};
+  & + li {
+    padding-top: ${space(4)};
+    border-top: 1px solid ${p => p.theme.borderLight};
+  }
+`;
+
+const Header = styled('div')`
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  margin-bottom: ${space(2)};
+`;
+
+const GroupVariantTitle = styled('h5')`
+  font-size: ${p => p.theme.fontSizeMedium};
+  margin: 0;
+`;
+
+const ContributionIcon = styled(({isContributing, ...p}) =>
+  isContributing ? (
+    <IconCheckmark size="sm" isCircled color={theme.green} {...p} />
+  ) : (
+    <IconClose size="sm" isCircled color={theme.red} {...p} />
+  )
+)`
+  margin-right: ${space(1)};
+  transform: translateY(3px);
```

Comment:
Can we make the container flex + `align-items: center` instead?

Reply:
Done.